openedx/core/djangoapps/courseware_api/serializers.py

```diff
@@ -115,6 +115,34 @@ class CourseInfoSerializer(serializers.Serializer):  # pylint: disable=abstract-
     is_integrity_signature_enabled = serializers.BooleanField()
     user_needs_integrity_signature = serializers.BooleanField()
     learning_assistant_enabled = serializers.BooleanField()
+    show_courseware_link = serializers.BooleanField()
+    is_course_full = serializers.BooleanField()
+    can_enroll = serializers.BooleanField()
+    invitation_only = serializers.BooleanField()
+    is_shib_course = serializers.BooleanField()
+    allow_anonymous = serializers.BooleanField()
+    ecommerce_checkout = serializers.BooleanField()
+    single_paid_mode = serializers.DictField()
+    ecommerce_checkout_link = AbsoluteURLField()
+    course_image_urls = serializers.ListField(
+        child=serializers.CharField(),
+        allow_empty=True,
+        default=list,
+    )
+    start_date_is_still_default = serializers.BooleanField()
+    advertised_start = serializers.CharField()
+    course_price = serializers.CharField()
+    pre_requisite_courses = serializers.ListField(
+        child=serializers.CharField(),
```

Comment:
Can we add a `PrerequisiteCourseSerializer` that will return serialized data instead of a string with a tuple?

```
class PrerequisiteCourseSerializer(serializers.Serializer):
    """
    Serializer for prerequisite course data with key and display name.
    """
    key = serializers.CharField()
    display = serializers.CharField()
```

```suggestion
        child=PrerequisiteCourseSerializer(),
```

Reply:
added

Reply:
Resolved